### PR TITLE
PXP-7499: Resolve regardless of the prefix when using /index?ids

### PR DESCRIPTION
**Pull request by MaribelleHGomez**
### Improvements

- Modified get_with_nonstrict prefix route (https://github.com/uc-cdis/indexd/blob/master/indexd/index/drivers/alchemy.py#L1101) to replace hardcoded prefix delimiter.
- (https://github.com/uc-cdis/indexd/blob/master/indexd/index/drivers/alchemy.py#L500) expanded logic to query additional ids by adding to id array. If the default prefix is not _None_ the id with default prefix will be appended to id array; similarly, following logic from the get_with_nonstrict prefix route, an id stripped of the default prefix will be added to the id array.
- Added test for get without prefix (https://github.com/uc-cdis/indexd/blob/chore/update_id_get/tests/test_client.py#L86)

### CODE REVIEW

**Comment by MaribelleHGomez** [comment]
I ran tests locally but I'd like some more feedback on unit tests

**Comment by BinamB** [comment]
> I ran some tests locally but I'd like some more feedback on unit tests
> These are the two tests for that area of code:
> `data_by_ids = client.get("/index/?ids={}".format(rec_1["did"])) assert data_by_ids.status_code == 200 data_list_all = data_by_ids.json`
> 
> ```
> `# with nonstrict prefix
> stripped = rec_1["did"].split("testprefix:", 1)[1]
> data_by_ids = client.get("/index/?ids={}".format(stripped))
> assert data_by_ids.status_code == 200
> data_list_all = data_by_ids.json`
> ```
> 
> also, some feedback on this (https://github.com/uc-cdis/indexd/blob/chore/update_id_get/indexd/index/drivers/alchemy.py#L511), ty

File: tests/test_client.py [hunk 1]
```diff
@@ -83,6 +83,12 @@ def test_index_list_with_params(client, user):
     assert data_by_ids.status_code == 200
     data_list_all = data_by_ids.json
```
Comment (BinamB) [inline tests/test_client.py:84]: This isn't being tested anymore since in ln 90 `data_list_all` replaces this one.

Comment (BinamB) [reply tests/test_client.py:84]: So i  guess i would suggest add the same types of checks in line 92 over here.

File: tests/test_client.py [hunk 2]
```diff
@@ -83,6 +83,12 @@ def test_index_list_with_params(client, user):
...
 
+    # with nonstrict prefix
+    stripped = rec_1["did"].split("testprefix:", 1)[1]
+    data_by_ids = client.get("/index/?ids={}".format(stripped))
+    assert data_by_ids.status_code == 200
+    data_list_all = data_by_ids.json
+
```
Comment (BinamB) [inline tests/test_client.py:91]: I guess I would also add different types of test cases like the one we talked about in chat where a guid and a pref+guid might exist. That is something we definitely want to handle.

**Review by BinamB — CHANGES_REQUESTED** [review]
I think its good. Just a small comment. Could you also remind me what Pauline wanted changed as well?

**Review by BinamB — APPROVED** [review]

**Review by BinamB — APPROVED** [review]

File: indexd/index/drivers/alchemy.py [hunk 3]
```diff
@@ -498,7 +498,25 @@ def ids(
                 query = query.order_by(IndexRecord.did)
 
             if ids:
-                query = query.filter(IndexRecord.did.in_(ids))
+                DEFAULT_PREFIX = self.config.get("DEFAULT_PREFIX")
+                found_ids = []
+                new_ids = []
+
+                if not DEFAULT_PREFIX:
+                    print("NO DEFAULT_PREFIX")
```
Comment (BinamB) [inline indexd/index/drivers/alchemy.py:506]: can you log this instead of print

**Review by BinamB — CHANGES_REQUESTED** [review]

**Review by BinamB — APPROVED** [review]

**Pull request merged**